web/rainmaker/dev-packages/egov-rented-properties-dev/src/ui-config/screens/specs/rented-properties/applyResource/propertyDetails.js

```diff
@@ -64,68 +64,41 @@ const colonyField = {
         dispatch(prepareFinalObject("colonySelectedCode", colonySelectedCode))
         dispatch(prepareFinalObject("applyScreenMdmsData.propertyAreas", propertyAreas))
         dispatch(prepareFinalObject("Properties[0].propertyDetails.rentPerSqyd", rentPerSqyd))
-        let interestRatePerYear = get(state.screenConfiguration.screenConfig["apply"],"components.div.children.formwizardFirstStep.children.rentDetails.children.cardContent.children.detailsContainer.children.interestRatePerYear.props.value");
-        let rentIncrementPercentage = get(state.screenConfiguration.screenConfig["apply"],"components.div.children.formwizardFirstStep.children.rentDetails.children.cardContent.children.detailsContainer.children.rentIncrementPercentage.props.value");
-        let rentIncrementPeriod = get(state.screenConfiguration.screenConfig["apply"], "components.div.children.formwizardFirstStep.children.rentDetails.children.cardContent.children.detailsContainer.children.rentIncrementPeriod.props.value");
-
-        // compare mdms colony data values for rent with field values if same dispatch 
-        //if not same dont dispatch
-
+        
+        let currentColony = get(state.screenConfiguration.preparedFinalObject,"Properties[0].colony");
         rentedPropertyColonies.map(item => {
```

Comment:
map is no need here

Reply:
if (action.value === item.code) {
for this I need map
else the change in colony is not changing
rent details